assignment-0/pkg/controller/controller.go

```diff
@@ -0,0 +1,154 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+type Controller struct {
+	Client        *kubernetes.Clientset
+	Informer      cache.SharedIndexInformer
+	Queue         workqueue.RateLimitingInterface
+	ServiceDelete bool
```

Comment:
can you add comment on what exactly `ServiceDelete` does?

Reply:
added